Symfony/src/AppBundle/Controller/UploadController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use AppBundle\Form\Type\FirmwareType;
+use AppBundle\Entity\FirmwareConfig;
+use Symfony\Component\HttpFoundation\Request;
+
+class UploadController extends Controller
+{
+    /**
+     * @Route("/upload")
+     */
+    public function showAction(Request $request)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+        $firmware = new FirmwareConfig();
+        $user = $this->getUser();
+        $firmware->setOwner($user?$user->getId():0);
+        $firmware->setUniqueUrl(base_convert(microtime(false), 10, 36));
+        $form = $this->createForm(FirmwareType::class, $firmware);
+        $form->handleRequest($request);
+        $errors = $this->get('validator')->validate($firmware);
+        if ($form->isSubmitted() && $form->isValid() && count($errors) == 0) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($firmware);
+            $entityManager->flush();
+            ?>
+            <script type="text/javascript">
```

Comment:
Ouch! This breaks the MVC model completely! We're not supposed to mix presentation with logic and definitely not output html from our controller, it can go wrong a thousand ways!

Also JS `alerts()` shouldn't really be used for anything, they are intrusive and block everything on the page. What you want to do instead is most probably use a flash message and a redirect.